### PR TITLE
Add the --tag flag to skaffold build

### LINKED ISSUE #3163: Add the --tag flag to skaffold build
### Expected behavior

Our usecase is as follows, we need to build and push the image with a calculated tag to a repository without deploying it immediately. The `skaffold build` command currently does not support the `--tag` flag same as the `skaffold run`, which makes our desired behavior quite impossible without generating a `skaffold.yml` for each build. 

### Actual behavior

While the `skaffold run` command supports the flag, `skaffold build` does not.

### Information

- Skaffold version: v0.41.0
- Operating system: OSX

### PR DESCRIPTION
**Pull request by oke-py**
Fixes #3163. 

**Description**

Add the --tag flag to skaffold build. See more details in #3163.

**Before**

No --tag flag to skaffold build

**After**

Add the --tag flag to skaffold build

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [x] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

```
Add the --tag flag to skaffold build
```

### CODE REVIEW

File: cmd/skaffold/app/cmd/build_test.go [hunk 1]
```diff
@@ -48,6 +48,40 @@ func (r *mockRunner) Stop() error {
 	return nil
 }
 
+func TestTagFlag(t *testing.T) {
+	mockCreateRunner := func(config.SkaffoldOptions) (runner.Runner, *latest.SkaffoldConfig, error) {
+		return &mockRunner{}, &latest.SkaffoldConfig{}, nil
+	}
+
+	tests := []struct {
```
Comment (dgageot) [inline cmd/skaffold/app/cmd/build_test.go]: Since there's a single test, you should be able to inline all those values.

**Comment by oke-py** [comment]
Running test locally succeeds but CI fails...

**Comment by oke-py** [comment]
`make generate-schemas` doesn't update any files.

File: cmd/skaffold/app/cmd/build_test.go (hunk 1 above)
Comment (oke-py) [reply cmd/skaffold/app/cmd/build_test.go]: I modified code and CI passed again.

File: cmd/skaffold/app/cmd/build.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ func NewCmdBuild() *cobra.Command {
 		WithCommonFlags().
 		WithFlags(func(f *pflag.FlagSet) {
 			f.StringSliceVarP(&opts.TargetImages, "build-image", "b", nil, "Choose which artifacts to build. Artifacts with image names that contain the expression will be built only. Default is to build sources for all artifacts")
+			f.StringVarP(&opts.CustomTag, "tag", "t", "", "The optional custom tag to use for images which overrides the current Tagger configuration")
```
Comment (tejal29) [inline cmd/skaffold/app/cmd/build.go]: This should go in https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/flags.go and should be registered for "run", "build", "dev" and "debug"

I think this cycle we are working on Multiple taggers.
Its strange, we only had this for Run and not for dev and debug which also runs skaffold build.

I am curious are you planning to use this in your CI/CD pipeline?

Comment (tejal29) [reply cmd/skaffold/app/cmd/build.go]: Read the linked issue.

Comment (tejal29) [reply cmd/skaffold/app/cmd/build.go]: can move this to flags.go and register it for `build` and `run` only for now.

**Pull request merged**